fractalis/data/etls/transmart/handler_transmart.py

```diff
@@ -56,9 +57,82 @@ def _get_token_for_credentials(self, server: str, auth: dict) -> str:
             raise ValueError(error)
         try:
             response = r.json()
-            return response['access_token']
         except Exception:
             error = "Could not authenticate. " \
                     "Got unexpected response: '{}'".format(r.text)
             logger.error(error)
             raise ValueError(error)
+        return response['access_token']
+
+    @staticmethod
+    def retrieve_keycloak_public_key_and_algorithm(oidc_server_url: str) -> (str, str):
+        handle = f'{oidc_server_url}/certs'
+        logger.info('Validating the token...')
+        r = requests.get(handle)
+        if r.status_code != 200:
+            error = "Could not validate the token. " \
+                    "Reason: [{}]: {}".format(r.status_code, r.text)
+            logger.error(error)
+            raise ValueError(error)
+        try:
+            json_response = r.json()
+            key0 = json_response.get('keys')[0]
```

Comment:
If there is only one this works, but if there are multiple? Should we filter based on the `kid` field in the token? I don't know in what circumstances there can be multiple.

Reply:
you are right, we should not assume there is always one, I'll change this part - indeed matching the kid with the one that is in the token header sounds like the best option. This was because currently we always use one, not sure why we would want to have multiple, but I just found the option to generate additional ones in keycloak.